src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -170,11 +173,18 @@ public virtual async Task OnAuthorizationAsync(AuthorizationFilterContext contex
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (context.HttpContext.Items.ContainsKey(AuthorizationMiddlewareInvokedKey))
```

Comment:
Is it worth having extension methods i.e. `context.HttpContext.HasAuthorization/CorsMiddlewareInvoked()` so these keys can be implementation details in Cors/AuthZ which can define the methods?

Reply:
I'm not sure who would use it other than us. It is always something we can do in the future if other people start checking for this key.

Reply:
Even if its just for us would an extension method be slightly cleaner than defining the same const in two places?

```
// Property key set by authorization middleware when it is run
        private const string AuthorizationMiddlewareInvokedKey = "__AuthorizationMiddlewareInvoked";
```

Reply:
Duplicating consts is not ideal. On the other hand I'm a hesitant about exposing a public extension method that blesses this behavior as official, forcing us to consider the API for future support and changes.

@HaoK/@rynowak do you know of any equivalent of this in ASP.NET Core today?

```cs
public bool HasAuthorizationInvoked(this HttpContext context)
{
    return context.Items.ContainsKey("__AuthKey");
}

public bool HasCorsInvoked(this HttpContext context)
{
    return context.Items.ContainsKey("__CorsKey");
}
```

Reply:
IMO this is like posting on facebook about relationship drama.

Reply:
Yep not blocking by any means